stubparser/src/org/checkerframework/stubparser/ast/visitor/DumpVisitor.java

```diff
@@ -221,9 +221,22 @@ private void printMembers(List<BodyDeclaration> members, Object arg) {
 
     private void printMemberAnnotations(List<AnnotationExpr> annotations, Object arg) {
         if (annotations != null) {
+            for (Iterator<AnnotationExpr> i = annotations.iterator(); i.hasNext();) {
+                AnnotationExpr a = i.next();
+                a.accept(this, arg);
+                if (i.hasNext()) {
+                    printer.print(" ");
+                }
+            }
+            printer.printLn();
+        }
+    }
+
+    private void printReceiverAnnotations(List<AnnotationExpr> annotations, Object arg) {

```

Comment:
This method isn't even documented.  Can you add that?  You said that the treatment of whitespace is the main difference among the various print*Annotations methods, but a reader is forced to reverse-engineer the spec of every routine.  If there was a spec, the code would be much easier to understand.  So, please make sure that every method has documentation, throughout the code.